cmd/convox/stdcli/stdcli.go

```diff
@@ -309,3 +306,51 @@ func ParseOpts(args []string) map[string]string {
 
 	return options
 }
+
+func NeedHelp(c *cli.Context) {
+	if c.IsSet("h") || c.GlobalIsSet("h") || c.IsSet("help") || c.GlobalIsSet("help") || c.Bool("h") || c.Bool("help") {
+		cli.ShowSubcommandHelp(c)
+		Exiter(0)
+	}
+
+	for _, a := range c.Args() {
+		if a == string('h') || a == "help" {
+			cli.ShowSubcommandHelp(c)
+			Exiter(0)
+		}
+	}
+}
+
+// NeedArg checks that we have received the expected number of arguments
+func NeedArg(c *cli.Context, count int) {
+	// if count is negative, it means we need *at least* one argument
+	if count < 0 && len(c.Args()) >= 1 {
+		return
+	}
+
+	expected := "argument"
+	if count > 1 || count == 0 {
+		expected = "arguments"
+	}
+
+	if len(c.Args()) < count || count < 0 {
+		Errorf("missing required %s %s",
+			expected,
+			c.Command.ArgsUsage)
+		Usage(c, c.Command.Name)
+	}
+
+	if len(c.Args()) > count {
+		msg := fmt.Sprintf("expected %d %s %s; got '%s'. Perhaps you meant to use a subcommand or option?",
```

Comment:
My first impression of this output is that it's redundant, because the CLI usage that's printed out should imply the problem.

```
$ git init foo bar
usage: git init [-q | --quiet] [--bare] [--template=<template-directory>] [--shared[=<permissions>]] [<directory>]
```

I also like it when we the CLI can hint at something more specific:

```
$ heroku app info
 !    `app` is not a heroku command.
 !    Perhaps you meant `api` or `apps`.
 !    See `heroku help` for a list of available commands.
```

However I do like:

* The consistency
* Tests that assert the stderr output